api/turing/api/appcontext.go

```diff
@@ -114,19 +114,22 @@ func NewAppContext(
 			mlpSvc,
 		)
 
-		batchClusterController, ok := clusterControllers[cfg.BatchEnsemblingConfig.JobConfig.DefaultEnvironment]
+		imageBuildingController, ok := clusterControllers[cfg.EnsemblerServiceBuilderConfig.ClusterName]
```

Comment:
I find it a little strange that the `batchEnsemblingJobRunner` would have its `imageBuildingController` be built using a cluster controller that's derived from the `EnsemblerServiceBuilderConfig`. Not a huge priority or a big problem but would it be better to move the `ClusterName` config to be under `ImageBuildingConfig`?

i.e. 
```yaml
BatchEnsemblingConfig:
  ImageBuildingConfig: &imageBuildingConfig
    ClusterName: build-cluster
    BuildNamespace: build-namespace
    ...
```
such that we get the same cluster name automatically copied over to the ensembler service builder config too?  
```yaml
EnsemblerServiceBuilderConfig:
  ImageBuildingConfig:
  <<: *imageBuildingConfig
  ...
```

Reply:
Given the current usage, I agree that the `ClusterName` can be embedded into the `ImageBuildingConfig` which also makes the configs more concise. Will address this in the next MR. Thanks for raising.